src/apps/material-grid/manual/MaterialGridManual.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { useEffect } from "react";
+import { WorkId } from "../../../core/utils/types/ids";
+import MaterialGrid from "../MaterialGrid";
+import { ValidSelectedIncrements } from "../materiel-grid-util";
+import MaterialGridSkeleton from "../MaterialGridSkeleton";
+
+export type MaterialGridManualProps = {
+  materialIds: WorkId[];
+  title: string;
+  selectedAmountOfMaterialsForDisplay: ValidSelectedIncrements;
+};
+
+const MaterialGridManual: React.FC<MaterialGridManualProps> = ({
+  materialIds,
+  title,
+  selectedAmountOfMaterialsForDisplay
+}) => {
+  const [parsedMaterialIds, setParsedMaterials] =
+    React.useState<WorkId[]>(materialIds);
+
+  useEffect(() => {
+    // Materials must be parsed from string to WorkId[] array as they are
+    // received as a string from the backend.
+    if (typeof materialIds === "string") {
+      setParsedMaterials(JSON.parse(materialIds));
+    }
+  }, [materialIds]);
+
+  if (!parsedMaterialIds.length || typeof parsedMaterialIds === "string") {
```

Comment:
Consider moving this handling of `materialIds`

I do not understand how TypeScript will accept a seemingly irrelevant type check for `materialIds` which we have declared should be `WorkId[]`.

I suggest moving the handling to the `.entry.tsx` file which is intended for CMS/non-Storybook usage. Here we can correctly declare the type of `materialIds` to be a string and then parse it to be an array of material ids we can pass along to the shared `MaterialGridManual` component.

Reply:
Makese sense. 

I refactored and moved the logic to the entry file. Have a look 🚀